src/stdx/bounded_array.zig

```diff
@@ -0,0 +1,72 @@
+const std = @import("std");
+const assert = std.debug.assert;
+
+/// A version of `std.BoundedArray` with TigerBeetle-idiomatic APIs.
```

Comment:
Maybe comment that the impetus for this wrapper is to add `fn count() usize` since `BoundedArray.len`'s type isn't a `usize`?

---

Also, would it be preferable to just use Zig's APIs for functions that are trivial wrappers? e.g. `appendAssumeCapacity`. That way if (hopefully) zig adds `count()` to its stdlib, it is trivial to remove the wrapper. (Similar to un-vendoring hashmap.)

Reply:
Yeah, that's a design question! My thinking here is that we don't actually want to remove this wrapper at least until Zig std is stable --- wrapper will help us with, eg, Zig version upgrades, by decoupling changes in the std from changes in our code (eg, if a method is renamed in a new version of Zig, its enough to rename just the wrapper, not every usage). The overall guideline I have in mind here is:

- we wrap std types to fix API errors, or otherwise tweak them for our
  particular usage
- we do this lazily, only when there's an actual problem
- we do this minimally (wrap API rather than vendor the whole copy)
- we _might_ unfork once Zig stdlib is stable (and switches to
  snake_case :-))

Reply:
> - we do this lazily, only when there's an actual problem
> - we do this minimally (wrap API rather than vendor the whole copy)

100% agree! That's why I suggested mirroring the underlying function names. :wink: 
I don't feel strongly about it though.